File: LeetCode/1221. 分割平衡字符串.cpp
class Solution
{
public:
    static int balancedStringSplit(const string &s)
    {
        int cnt = 0, flag = 0;
        for (auto i: s)
        {
            i == 'L' ? flag += 1 : flag -= 1;
            cnt = flag == 0 ? cnt + 1 : cnt;
        }
        return cnt;
    }
};

File: LeetCode/1221-分割平衡字符串.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        cnt = 0
        flag = 0

        for i in s:
            if i == "L":
                flag += 1
            else:
                flag -= 1

            if flag == 0:
                cnt += 1

        return cnt


if __name__ == "__main__":
    obj = Solution()
    print(obj.balancedStringSplit("RLRRLLRLRL"))
